### PR TITLE
dm(engine): fix some problem and add case sensitive test

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/6776

### What is changed and how it works?

fix problems:

- case-sensitive is not working
- dump unit tries to remove wrong directory

#### Tests 

 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by lance6716** [comment]
[df.log](https://github.com/pingcap/tiflow/files/9350564/df.log)

log of 

```
3m3e-server-executor-1               | [2022/08/16 11:08:10.569 +00:00] [WARN] [writer_util.go:458] ["fail to open file"] [job_id=93642358-bbf6-43d2-807a-aad17c4299f2] [worker_id=8b8729c3-0186-4a2d-bab1-8bedc5759eef] [task=test] [unit=dump] [path=file:////8b8729c3-0186-4a2d-bab1-8bedc5759eef/746573742f6d7973716c2d3032/lower_db-schema-create.sql] [error="open /8b8729c3-0186-4a2d-bab1-8bedc5759eef/746573742f6d7973716c2d3032/lower_db-schema-create.sql: no such file or directory"]
```

@liuzix

**Comment by lance6716** [comment]
inside executor:

```
/ # ls /tmp/dfe-storage/6578656375746f722d7365727665722d6578656375746f722d313a3130323431/
5640d006-626f-47a3-ac6f-4f7c37384785
/ # ls /tmp/dfe-storage/6578656375746f722d7365727665722d6578656375746f722d313a3130323431/5640d006-626f-47a3-ac6f-4f7c37384785/
```

But dumpling requires `"RawCause": "open /tmp/dfe-storage/6578656375746f722d7365727665722d6578656375746f722d313a3130323431/5640d006-626f-47a3-ac6f-4f7c37384785/746573742f6d7973716c2d3032/lower_db-schema-create.sql: no such file or directory"`

File: engine/jobmaster/dm/config/config.go [hunk 1]
```diff
@@ -257,7 +258,7 @@ func (c *TaskCfg) ToDMSubTaskCfg() *dmconfig.SubTaskConfig {
 	cfg.OnlineDDL = c.OnlineDDL
 	cfg.ShadowTableRules = c.ShadowTableRules
 	cfg.TrashTableRules = c.TrashTableRules
-	cfg.CaseSensitive = c.CaseSensitive
+	cfg.CaseSensitive = c.Upstreams[0].CaseSensitive
```
Comment (GMHDBJD) [inline engine/jobmaster/dm/config/config.go:261]: better move to L278

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Comment by lance6716** [comment]
ptal @maxshuang @CharlesCheung96

**Review by CharlesCheung96 — APPROVED** [review]

File: engine/test/utils/run_sql_file [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
```
Comment (amyangfei) [inline engine/test/utils/run_sql_file:1]: We can reuse test scripts later, are these files duplicated
- dm/tests/_utils/run_sql_file
- tests/integration_tests/_utils/run_sql_file

**Pull request merged**